tests/instruments/agilent/test_agilent4294A.py

```diff
@@ -0,0 +1,49 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import pytest
+from pymeasure.test import expected_protocol
+from pymeasure.instruments.agilent.agilent4294A import Agilent4294A
+
+
+@pytest.mark.parametrize(
+    "freq", [40, 140E6]
+)
+def test_frequency(freq):
+    """
+    Test Agilent 4294A stimulus frequency
+    """
+    with expected_protocol(
+        Agilent4294A,
+        [
+            ("STAR?", freq),
+            ("STOP?", freq),
+            (f"STAR {freq:.0f} HZ", None),
+            (f"STOP {freq:.0f} HZ", None),
+        ],
+    ) as inst:
+        assert freq == inst.start_frequency
+        assert freq == inst.stop_frequency
```

Comment:
As you test different properties, you could move them to different tests, one for each property.
You could even split the getter and setter part of the properties into separate tests.